temporalio/workflow.py

```diff
@@ -436,6 +446,20 @@ async def workflow_wait_condition(
         ...
 
 
+def deprecate_patch(id: str) -> None:
+    """Mark a patch as deprecated.
+
+    This marks a workflow that had :py:func:`patched` in a previous version of
+    the code as no longer applicable because all workflows that use the old code
+    path are done and will never be queried again. Therefore the old code path
+    is removed as well.
```

Comment:
Not necessary for this PR, but probably makes sense to link to docs page explaining the patch lifecycle

Reply:
Unfortunately that doesn't exist for Python yet. https://docs.temporal.io/typescript/patching will be perfect in Python form if/when it exists. I have opened #37 to track this.

Reply:
It should probably be part of the app dev guide